src/shogun/regression/GLM.cpp

```diff
@@ -0,0 +1,80 @@
+#include <shogun/lib/config.h>
+
+#include <shogun/labels/RegressionLabels.h>
+#include <shogun/lib/SGVector.h>
+#include <shogun/mathematics/Math.h>
+#include <shogun/mathematics/linalg/LinalgNamespace.h>
+#include <shogun/regression/GLM.h>
+#include <utility>
+using namespace shogun;
+
+GLM::GLM() : LinearMachine()
+{
+	init();
+}
+SGVector<float64_t> log_likelihood(
+    const std::shared_ptr<DenseFeatures<float64_t>>& features,
+    const std::shared_ptr<Labels>& label)
+{
+	auto vector_count = features->get_num_vectors();
+	auto feature_count = features->get_num_features();
+	ASSERT(vector_count > 0 && label->get_num_labels() == vector_count)
+	// Array of Lambdas
+	SGVector<float64_t> lambda(vector_count);
+	for (auto i = 0; i < vector_count; i++)
+	{
+		SGVector<float64_t> feature_vector = features->get_feature_vector(i);
+		// Assume beta is the same as the feature vector
+		SGVector<float64_t> beta = feature_vector.clone();
+		// Assume beta0 is the same as the first element in the feature vector
+		float64_t b0 = feature_vector.get_element(0);
+		float64_t res = linalg::dot(feature_vector, beta);
+		lambda.set_element(log(1 + std::exp(b0 + res)), i);
+	}
+	SGVector<float64_t> likelihood(vector_count);
+	SGVector<float64_t> labels = label->get_values();
+	SGVector<float64_t> log_lambda(vector_count);
+
+	for (auto i = 0; i < vector_count; i++)
+		log_lambda.set_element(log(lambda.get_element(i)), i);
+
+	likelihood = linalg::add(
+	    linalg::element_prod(labels, log_lambda), lambda, 1.0, -1.0);
+	SGVector<float64_t> likelihood_clone = likelihood.clone();
+	for (auto i = 0; i < vector_count; i++)
+		likelihood.set_element(
+		    SGVector<float64_t>::sum(likelihood_clone, i), i);
+	return likelihood;
+}
+void GLM::init()
+{
+	SG_ADD(
```

Comment:
this is for the elastic net stuff, which we don't need here for now. Just the L2 reguariser is enough

Reply:
so just removing alpha and lambda and adding tau as in `LinearRidgeRegression` ?

Reply:
Yes (or use lambda instead of tau). And please add "L2" to the description to clarify it's a ridge penalty

Reply:
Done, I think I will use tau for this method to keep it consistent with the other models such as LinearRidgeRegression + I am using the naming lambda for the likelihood array